hail/python/test/hail/matrixtable/test_matrix_table.py

```diff
@@ -2383,3 +2383,145 @@ def test_struct_of_arrays_encoding():
         etype = md['_codecSpec']['_eType']
         assert 'EStructOfArrays' in etype
         assert mt._same(std_mt)
+
+
+def test_query_matrix_table():
+    n_cols = 100
+    f = new_temp_file(extension='mt')
+    mt = hl.utils.range_matrix_table(n_rows=200, n_cols=n_cols, n_partitions=10)
+    mt = mt.filter_rows(mt.row_idx % 10 == 0)
+    mt = mt.filter_cols(mt.col_idx % 10 == 0)
+    mt = mt.annotate_rows(s=hl.str(mt.row_idx))
+    mt = mt.annotate_entries(n=mt.row_idx * mt.col_idx)
+    mt.write(f)
+
+    queries = [
+        hl.query_matrix_table(f, 50, 'e'),
+        hl.query_matrix_table(f, hl.struct(idx=50), 'e'),
+        hl.query_matrix_table(f, 55, 'e'),
+        hl.query_matrix_table(f, 5, 'e'),
+        hl.query_matrix_table(f, -1, 'e'),
+        hl.query_matrix_table(f, 205, 'e'),
+        hl.query_matrix_table(f, hl.interval(27, 66), 'e'),
+        hl.query_matrix_table(f, hl.interval(276, 33333), 'e'),
+        hl.query_matrix_table(f, hl.interval(-22276, -5), 'e'),
+        hl.query_matrix_table(f, hl.interval(hl.struct(row_idx=27), hl.struct(row_idx=66)), 'e'),
+        hl.query_matrix_table(f, hl.interval(40, 80, includes_end=True), 'e'),
+    ]
```

Comment:
Can you make this a parameterised test please - if one of these fail then we won't know if all or just that particular case is borked

Reply:
Sure. I'll do the `query_table` tests while I'm at it.